corehq/sql_db/util.py

```diff
@@ -49,6 +50,34 @@ def run_query_across_partitioned_databases(model_class, q_expression, values=Non
             yield result
 
 
+def get_objects_by_id_across_partitioned_databases(model_class, model_ids, additional_query_params=None):
+    """
+    Runs a query across all partitioned databases and produces a generator
+    with the results.
+
+    :param model_class: A Django model class
+
+    :param model_ids: The list of IDs to retreive
+
+    :param q_expression: An instance of django.db.models.Q representing any additional
+    filters to apply (optional)
+
+    :return: A generator with the results
+    """
+    assert issubclass(model_class, PartitionedModel)
+    assert model_class.partition_attr
+    query_param = '{}__in'.format(model_class.partition_attr)
```

Comment:
Could use `Q` objects here instead of a dict. That would also enable passing arbitrary other filters in (similar to `run_query_across_partitioned_databases`)

Reply:
we also have to keep in mind that `partition_attr` is just the attribute we use to partition the object, but not necessarily the primary key we use to do lookups.

for example, some models may be sharded by case_id but have a different primary key that we use to lookup the actual objects.

might be good to rename `model_ids` to `partition_ids`, rename the function to `get_objects_by_partition_id_across_partitioned_databases`, and update the docstring to explain this difference